### PR TITLE
docs(plan): enforce ready-for-regression label and reviewer-loop handoff via CI

### PR DESCRIPTION
**Pull request by lhpaul**
## Summary

Implementation plan for issue #613 — structural CI enforcement of two recurring workflow gaps:
1. `ready-for-regression` label missing when agents declare PRs ready for human review.
2. PRs treated as ready without the automated reviewer-loop running to completion.

## Approach

**Two new GitHub Actions workflow files**:
- `.github/workflows/apply-regression-label.yml` — triggers on `pull_request` events (opened, reopened, ready_for_review, synchronize); auto-applies `ready-for-regression` to implementation branches (`feature/`, `fix/`, `refactor/`, `hotfix/`); skips all other branch prefixes; idempotent.
- `.github/workflows/reviewer-loop-guard.yml` — triggers on the same events; asserts the canonical `### Automated Reviewer Loop Summary` comment is present on the PR; posts a GitHub commit status (`Reviewer-loop completion guard`) that downstream repos can add to branch protection as a required check.

**One new integration doc**: `docs/workflow/development-workflow/integrations/ci-enforcement.md` explaining default branch prefix list, downstream override, and branch protection wiring.

**Estimated complexity**: S — two self-contained YAML files and a doc page. No existing code is modified; no secrets or external services required beyond `GITHUB_TOKEN`.

**Key risks**:
- `ready-for-regression` label might not exist in target repos → mitigated by `gh label create --force` in the workflow.
- Guard uses Commit Statuses API (not Checks API) → documented in smoke test troubleshooting.

## Links

- Spec: `docs/specs/developments/20260515162721_613-enforce-regression-label-and-reviewer-loop-via-ci/1_613-enforce-regression-label-and-reviewer-loop-via-ci_specs.md`
- Implementation plan: `docs/specs/developments/20260515162721_613-enforce-regression-label-and-reviewer-loop-via-ci/2_613-enforce-regression-label-and-reviewer-loop-via-ci_implementation-plan.md`
- Smoke test runbook: `docs/testing/workflow/613-enforce-regression-label-and-reviewer-loop-via-ci.smoke-test.md`

Closes #613 (plan stage only — issue remains open until implementation PR is merged).

## Summary by CodeRabbit

* **Documentation**
  * Added an implementation plan for CI-driven automation that applies a regression-readiness label to in-scope branches and gates merge readiness on a reviewer-loop completion check.
  * Added a smoke-test runbook with step-by-step validation, acceptance criteria, verification checklist, idempotency/edge-case handling, troubleshooting guidance, and cleanup instructions.

[Review Change Stack](https://app.coderabbit.ai/change-stack/lhpaul/ai-dev-framework-template/pull/641?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

**Comment by lhpaul** [comment]
### Automated Reviewer Loop Summary

**Result:** clean — no blocking findings
**Platforms:** pr-agent,coderabbit
**Findings:** 0 blocking, 0 suggestions

*Posted automatically by `pr-review-loop.sh`.*

File: docs/specs/developments/20260515162721_613-enforce-regression-label-and-reviewer-loop-via-ci/2_613-enforce-regression-label-and-reviewer-loop-via-ci_implementation-plan.md [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+## Verification Log
+
+| Check | Command / query | Result |
+| --- | --- | --- |
+| Repo revision | `git rev-parse --short HEAD` | `50b337d` |
+| Existing regression-label workflow count | `ls .github/workflows/ \| grep regression` | 1 (`remove-regression-label-on-push.yml`) |
+| Reviewer-loop summary marker (exact string) | `grep "Automated Reviewer Loop Summary" scripts/development-workflow/pr-review-loop.sh` | Present at lines 123, 3151, 3155, 3304, 3321, 3335 |
+| Secondary marker (unique to script-posted comments) | `grep "Posted automatically by" scripts/development-workflow/pr-review-loop.sh` | Present; combined with `### Automated Reviewer Loop Summary` to uniquely identify script-posted summaries |
+| Existing action pin (checkout) | `grep "actions/checkout" .github/workflows/remove-regression-label-on-push.yml` | `actions/checkout@93cb6efe18208431cddfb8368fd83d5badbf9bfd # v5.0.1` |
+| Existing in-scope branch prefixes in template | `grep -h "feature/\|fix/\|refactor/\|hotfix/" .github/workflows/remove-regression-label-on-push.yml` | Not listed in that file — prefixes are documented in spec; workflow defines them as a configurable variable |
```
[the remark this answers is not in the document]
Comment (lhpaul) [reply docs/specs/developments/20260515162721_613-enforce-regression-label-and-reviewer-loop-via-ci/2_613-enforce-regression-label-and-reviewer-loop-via-ci_implementation-plan.md]: Fixed: replaced the verbose text with the concise form "Not present (prefixes defined as env var in workflow)".

File: docs/specs/developments/20260515162721_613-enforce-regression-label-and-reviewer-loop-via-ci/2_613-enforce-regression-label-and-reviewer-loop-via-ci_implementation-plan.md [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+          gh label create "ready-for-regression" \
+            --repo "$REPO" --color "e4e669" \
+            --description "PR is ready for regression testing" \
+            --force || true
+          # Apply the label (idempotent via --add-label)
+          gh pr edit "$PR_NUMBER" \
+            --repo "$REPO" \
+            --add-label "ready-for-regression"
+          echo "Label 'ready-for-regression' applied to PR #${PR_NUMBER}."
+```
```
[the remark this answers is not in the document]
Comment (lhpaul) [reply docs/specs/developments/20260515162721_613-enforce-regression-label-and-reviewer-loop-via-ci/2_613-enforce-regression-label-and-reviewer-loop-via-ci_implementation-plan.md:143]: Fixed: added a `concurrency` block to the apply-regression-label code sample, keyed on `github.event.pull_request.number` with `cancel-in-progress: true`.

File: docs/specs/developments/20260515162721_613-enforce-regression-label-and-reviewer-loop-via-ci/2_613-enforce-regression-label-and-reviewer-loop-via-ci_implementation-plan.md [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+            DESCRIPTION="No reviewer-loop summary found. Run the automated reviewer loop before merging."
+          fi
+
+          gh api \
+            --method POST \
+            "repos/$REPO/statuses/$HEAD_SHA" \
+            -f state="$STATE" \
+            -f description="$DESCRIPTION" \
+            -f context="Reviewer-loop completion guard"
+```
```
[the remark this answers is not in the document]
Comment (lhpaul) [reply docs/specs/developments/20260515162721_613-enforce-regression-label-and-reviewer-loop-via-ci/2_613-enforce-regression-label-and-reviewer-loop-via-ci_implementation-plan.md:216]: Fixed: added a `concurrency` block keyed on PR number and added branch-prefix filtering (the same `IN_SCOPE_PREFIXES` check used in apply-regression-label) to the reviewer-loop-guard code sample. Out-of-scope branches now post a passing status and exit early.

File: docs/specs/developments/20260515162721_613-enforce-regression-label-and-reviewer-loop-via-ci/2_613-enforce-regression-label-and-reviewer-loop-via-ci_implementation-plan.md [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+     /repo/.github/workflows/reviewer-loop-guard.yml
+   ```
+
+   Fix any reported issues before committing.
+
+8. **Commit and push**: stage all four new/modified files:
+   - `.github/workflows/apply-regression-label.yml`
+   - `.github/workflows/reviewer-loop-guard.yml`
+   - `docs/workflow/development-workflow/integrations/ci-enforcement.md`
+   - `docs/specs/developments/20260515162721_613-enforce-regression-label-and-reviewer-loop-via-ci/2_613-enforce-regression-label-and-reviewer-loop-via-ci_implementation-plan.md` (plan — already on the plan branch; the impl branch will carry the actual workflow files)
```
[the remark this answers is not in the document]
Comment (lhpaul) [reply docs/specs/developments/20260515162721_613-enforce-regression-label-and-reviewer-loop-via-ci/2_613-enforce-regression-label-and-reviewer-loop-via-ci_implementation-plan.md]: Fixed: removed the plan file from the Step 8 commit list. The step now lists only the three implementation artifacts (`.github/workflows/apply-regression-label.yml`, `.github/workflows/reviewer-loop-guard.yml`, `docs/workflow/development-workflow/integrations/ci-enforcement.md`).

File: docs/testing/workflow/613-enforce-regression-label-and-reviewer-loop-via-ci.smoke-test.md [hunk 5]
```diff
@@ -0,0 +1,297 @@
...
+   ```bash
+   gh pr create \
+     --base develop \
+     --head fix/613-smoke-test-fixture \
+     --title "chore: smoke test fixture for #613 CI enforcement" \
+     --body "Smoke test fixture PR — delete after verification." \
+     --draft
+   ```
+
+3. Note the PR number: `PR_NUM=$(gh pr view --json number --jq '.number')`.
```
[the remark this answers is not in the document]
Comment (lhpaul) [reply docs/testing/workflow/613-enforce-regression-label-and-reviewer-loop-via-ci.smoke-test.md]: Fixed: both PR number capture steps (Step 1 for `PR_NUM` and Step 6 for `PR_NUM_2`) are now presented as fenced bash code blocks with the explicit variable assignment commands.

File: docs/testing/workflow/613-enforce-regression-label-and-reviewer-loop-via-ci.smoke-test.md [hunk 6]
```diff
@@ -0,0 +1,297 @@
...
+   HEAD_SHA=$(gh pr view "$PR_NUM" --json headRefOid --jq '.headRefOid')
+   gh api "repos/$(gh repo view --json nameWithOwner --jq '.nameWithOwner')/commits/$HEAD_SHA/statuses" \
+     --jq '[.[] | select(.context == "Reviewer-loop completion guard")] | last | {state, description}'
+   ```
+
+**Expected result**:
+
+- `state` is `"failure"` (guard re-evaluated after the push and the new SHA has no summary comment yet).
+- **Maps to**: Acceptance Criterion #7 (guard re-evaluates on every push).
+
```
[the remark this answers is not in the document]
Comment (lhpaul) [reply docs/testing/workflow/613-enforce-regression-label-and-reviewer-loop-via-ci.smoke-test.md]: Fixed: Step 5 heading and body updated to align with the comment-based contract. The step no longer asserts the guard will fail on the new SHA; instead it asserts that the guard re-evaluates and posts a fresh status reflecting whether the canonical summary comment is present. The AC #7 checkbox was updated accordingly. The reference to `remove-regression-label-on-push` was also removed from the wait instruction since it is not relevant to the guard re-evaluation check.

File: docs/testing/workflow/613-enforce-regression-label-and-reviewer-loop-via-ci.smoke-test.md [hunk 7]
```diff
@@ -0,0 +1,297 @@
...
+---
+
+## Seed Data Reference
+
+No seed data is required. The smoke test uses fixture branches and PRs created inline.
+
+| Entity | Scenario | How to load |
+| --- | --- | --- |
+| — | — | — |
+
```
[the remark this answers is not in the document]
Comment (lhpaul) [reply docs/testing/workflow/613-enforce-regression-label-and-reviewer-loop-via-ci.smoke-test.md]: Fixed: removed the empty placeholder table from the Seed Data Reference section in the smoke test. The explanatory sentence is retained. Also removed the equivalent empty table from the plan file's Seed Data section.

File: docs/specs/developments/20260515162721_613-enforce-regression-label-and-reviewer-loop-via-ci/2_613-enforce-regression-label-and-reviewer-loop-via-ci_implementation-plan.md [hunk 8]
```diff
@@ -0,0 +1,290 @@
...
+   - Match comments that contain **both** `### Automated Reviewer Loop Summary` and `*Posted automatically by \`pr-review-loop.sh\`.*` (same two-marker combination used in `pr-review-loop.sh` line 3335).
+   - Post a GitHub commit status on `HEAD_SHA` (`${{ github.event.pull_request.head.sha }}`): `success` when found, `failure` when absent.
+   - Use context string `"Reviewer-loop completion guard"` so downstream repos can add it to branch protection as a required check by name.
+   - Use `pull-requests: read` and `statuses: write` permissions (minimum required).
+   - Add a `concurrency` group keyed on the PR number (`reviewer-loop-guard-${{ github.event.pull_request.number }}`) with `cancel-in-progress: true`.
+
+4. **Create `docs/workflow/development-workflow/integrations/ci-enforcement.md`**: write a new integration doc that covers:
+   - What the two CI workflows do and when they run.
+   - Default in-scope branch prefixes (`feature/`, `fix/`, `refactor/`, `hotfix/`).
+   - How downstream repos override the prefix list (override the `IN_SCOPE_PREFIXES` environment variable in the workflow file, or fork and edit the `push` filter).
```
[the remark this answers is not in the document]
Comment (lhpaul) [reply docs/specs/developments/20260515162721_613-enforce-regression-label-and-reviewer-loop-via-ci/2_613-enforce-regression-label-and-reviewer-loop-via-ci_implementation-plan.md]: Fixed: replaced 'push filter' with 'pull_request trigger/scope logic' on line 245 of the plan. The two CI workflows in this plan are triggered by the `pull_request` event, so the override guidance now accurately points implementers to the correct trigger type. Commit d6df307.

File: docs/testing/workflow/613-enforce-regression-label-and-reviewer-loop-via-ci.smoke-test.md [hunk 9]
```diff
@@ -0,0 +1,302 @@
...
+### Step 2: Verify the reviewer-loop guard fails on the new PR
+
+1. Wait for the `reviewer-loop-guard` workflow run to complete on the fixture PR (typically < 60 s).
+2. Check the commit status:
+
+   ```bash
+   HEAD_SHA=$(gh pr view "$PR_NUM" --json headRefOid --jq '.headRefOid')
+   gh api "repos/$(gh repo view --json nameWithOwner --jq '.nameWithOwner')/commits/$HEAD_SHA/statuses" \
+     --jq '[.[] | select(.context == "Reviewer-loop completion guard")] | last | {state, description}'
+   ```
```
[the remark this answers is not in the document]
Comment (lhpaul) [reply docs/testing/workflow/613-enforce-regression-label-and-reviewer-loop-via-ci.smoke-test.md:79]: Fixed: replaced `| last` with `| first` in all three `gh api` status-check commands in the smoke test (Steps 2, 4, and 5). The GitHub REST API returns commit statuses in reverse chronological order (most recent first), so `first` correctly retrieves the latest status for the 'Reviewer-loop completion guard' context. Using `last` would have returned the oldest status, potentially causing incorrect test conclusions. Commit d6df307.
